pkg/revocation/revocation_db.go

```diff
@@ -1,44 +1,66 @@
 // Copyright (C) 2019 Storj Labs, Inc.
 // See LICENSE for copying information.
 
-package identity
+package revocation
 
 import (
 	"context"
 	"crypto/x509"
 	"crypto/x509/pkix"
 
+	"github.com/zeebo/errs"
+	"gopkg.in/spacemonkeygo/monkit.v2"
+
 	"storj.io/storj/internal/dbutil"
+	"storj.io/storj/pkg/identity"
 	"storj.io/storj/pkg/peertls"
 	"storj.io/storj/pkg/peertls/extensions"
+	"storj.io/storj/pkg/peertls/tlsopts"
 	"storj.io/storj/storage"
 	"storj.io/storj/storage/boltdb"
 	"storj.io/storj/storage/redis"
 )
 
-// RevocationDB stores the most recently seen revocation for each nodeID
+var (
+	mon = monkit.Package()
+
+	// Error is a pkg/revocation error
+	Error = errs.Class("revocation error")
+)
+
+// DB stores the most recently seen revocation for each nodeID
 // (i.e. nodeID [CA certificate's public key hash] is the key, values is
 // the most recently seen revocation).
-type RevocationDB struct {
-	DB storage.KeyValueStore
+type DB struct {
+	KVStore storage.KeyValueStore
+}
+
+// NewDBFromCfg is a convenience method to create a revocation DB
+// directly from a config. If the revocation extension option is not set, it
+// returns a nil db with no error.
+func NewDBFromCfg(cfg tlsopts.Config) (*DB, error) {
+	if !cfg.Extensions.Revocation {
+		return nil, nil
+	}
+	return NewDB(cfg.RevocationDBURL)
 }
 
-// NewRevocationDB returns a new revocation database given the URL
-func NewRevocationDB(revocationDBURL string) (*RevocationDB, error) {
-	driver, source, err := dbutil.SplitConnstr(revocationDBURL)
+// NewDB returns a new revocation database given the URL
+func NewDB(dbURL string) (*DB, error) {
```

Comment:
I would also split up bolt and redis implementations of revocation databases, so you can use one without the other.

Reply:
Are you saying so that, if the caller already knows that they are using bolt or redis, they can avoid importing the other? Because right now the constructor functions just take a string, so it seems any function with the signature `func NewDB(dbURL string) (*DB, error)` will need to import both.

Or am I misunderstanding?